### DISCUSSION TITLE
Need to Understand more about getCellsForSelection

### DISCUSSION THREAD

**Discussion by sudhansu36** [glideapps/glide-data-grid#530] [Q&A] [answered]
While I am using getCellsForSelection use Callback function it gives me the selection(x, y, width, height), but it only works till width 2 and if I go beyond width 2 its returning me width as 1 only. where my height is properly working. 
Can any one help me why this is happening.
  const getCellsForSelection = useCallback(
    (selection) => {
     //selection={x:1,y:1,width:1,height:1}
  
      return result;// result is List of selections
    },
    [getData]
  );

**Comment by jassmith** [comment]
getCellsForSelection is used in many locations and requests chunks based on its operational needs at the time. Often this is column by column.

**Reply by sudhansu36** [reply]
Can you please explain more about column by column.
Also If I want to select 4 rows and 4 columns will it work?
Because for me If I am selecting 3 columns and any no's of rows  its not providing me the correct result for my selection.

In my code I am getting selection as
{x: 8, y: 15, width: 1, height: 1}// for 1 cell selection
{x: 8, y: 15, width: 2, height: 1}// for 2 cell selection
{x: 8, y: 15, width: 1, height: 1}// for 3 cell selection
{x: 10, y: 15, width: 1, height: 1}// for 3 cell selection
{x: 8, y: 15, width: 1, height: 1}// for 4 cell selection
{x: 11, y: 15, width: 1, height: 1}// for 4 cell selection

Is this the actual behavior?

**Comment by jassmith** [comment, marked as answer]
This callback is not for telling you what the new selection is but is a
bull version of getCellContent. I think you are looking to use
onGridSelectionChange.

**Reply by sudhansu36** [reply]
Actually I am looking for the selected cell data while doing a range selection. So You can refer to this image.
![image](https://user-images.githubusercontent.com/49445013/195084436-68c23f6f-93ae-4fd8-8bd5-8751f002ce03.png)
I selected some rows and columns and wanted the cell coordinates for all selection, so which function I have to use?

**Reply by jassmith** [reply]
when `onGridSelectionChange` fires it gives you a `GridSelection` which has a current.cell and a current.range. Those ware what you want. Note that like an input if you hook up `onGridSelectionChange` you should also use `gridSelection` to pass back the new grid selection.

**Reply by jassmith** [reply]
To be slightly clearer, `gridSelection` is similar to `value` on an input, and `onGridSelectionChange` is similar to `onChange` on an input.
